.github/workflows/update-protobufs.yaml

```diff
@@ -0,0 +1,33 @@
+name: Update Protobufs
+on:
+  push:
+    branches:
+    - yaakov/auto-protos # temporary trigger during development
+  #schedule:
+  #- cron: '0 17 * * SUN'
+
+jobs:
+  update-protobufs:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2.3.4
+
+    - name: Run Update Script
+      shell: pwsh
+      run: |
+        git submodule init
+        Resources/ProtobufGen/update.ps1
+
+    # TODO: We need a new GitHub Machine Account (or maybe an existing one?) to generate a PAT that we can use as the token here
+    # otherwise, our new changes will not trigger further Actions (on:push or on:pull_request, i.e. CI/CD builds).
+    #- name: Create Pull Request
+    #  uses: peter-evans/create-pull-request@v3.10.1
```

Comment:
I haven't played with GH actions much - but what does security look like for a third-party action that is fed a PAT?

Does pinning the action to a specific version guarantee it to be immutable?

Reply:
Not sure that we have a machine account around (I definitely don't have one), so we'll want to create one - and share creds among the core team?

The PAT will definitely need to be limited in scope to just creating PRs, if that's doable.

Reply:
You can provide a full commit hash as the version afaik.

Reply:
It might be that pull requests created when run by cron shouldn't require a PAT.

> Note: If you want pull requests created by this action to trigger an on: push or on: pull_request workflow then you cannot use the default GITHUB_TOKEN.

FWIW, adding `permissions` section is worthwhile:
contents: write
pull-requests: write